src/ipc/ipc4/dai.c

```diff
@@ -84,6 +85,29 @@ int dai_config_dma_channel(struct dai_data *dd, struct comp_dev *dev, const void
 #endif
 		break;
 	case SOF_DAI_INTEL_HDA:
+#if defined(CONFIG_ACE_VERSION_2_0)
+		if (copier_cfg->gtw_cfg.node_id.f.dma_type == ipc4_alh_link_output_class ||
+		    copier_cfg->gtw_cfg.node_id.f.dma_type == ipc4_alh_link_input_class) {
+			struct processing_module *mod = comp_get_drvdata(dev);
+			struct copier_data *cd = module_get_private_data(mod);
+
+			if (!cd->gtw_cfg) {
+				comp_err(dev, "No gateway config found!");
+				return DMA_CHAN_INVALID;
+			}
+
+			channel = DMA_CHAN_INVALID;
+			const struct sof_alh_configuration_blob *alh_blob = cd->gtw_cfg;
+
+			for (int i = 0; i < alh_blob->alh_cfg.count; i++) {
+				if (dai->host_dma_config[i]->stream_id == dai->dai_index) {
+					channel = dai->host_dma_config[i]->dma_channel_id;
```

Comment:
same here, what does 'host_dma_config' mean in this context?

If it's meant to refer to the link DMA configuration setup by the host, it's really poor naming. All the cAVS/ACE documentation refers to host_dma by opposition to the link_dma. I am afraid the naming blurs the concepts at best, and is possibly wrong.

Reply:
`host_dma_config ` contains pointer to  `struct ipc_dma_config` and it's meant to refer to link DMA config. Please check my explanation above.